include/ginkgo/config.hpp.in

```diff
@@ -76,6 +76,12 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 // clang-format on
 
 
+/* What is the major version of dpcpp compiler */
+// clang-format off
+#define GINKGO_DPCPP_MAJOR_VERSION @GINKGO_DPCPP_MAJOR_VERSION@
```

Comment:
Could you describe it more detail?

Reply:
The config file is generated before `add_subdirectory(dpcpp)`. Therefore the code in the section below (using `__SYCL_COMPILER_VERSION`) will not populate the value in the configuration file? I guess the value will always be either 6 or 0 in this config.hpp file.

Reply:
it will populate the value in the config.hpp because we set CXX compiler to dpcpp already.
If we can split the sycl and core compiler, the `ginkgo_extract_dpcpp_version` will need to use the sycl compiler

Reply:
https://github.com/intel/llvm/pull/3300 for reference